contrib/cassandra-connector/src/main/java/com/google/cloud/cassandra/dataflow/io/CassandraReadIO.java

```diff
@@ -0,0 +1,188 @@
+package com.google.cloud.cassandra.dataflow.io;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import javax.swing.text.html.parser.Entity;
+
+import org.joda.time.Instant;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.mapping.Mapper;
+import com.datastax.driver.mapping.MappingManager;
+import com.google.cloud.dataflow.sdk.coders.Coder;
+import com.google.cloud.dataflow.sdk.coders.SerializableCoder;
+import com.google.cloud.dataflow.sdk.io.BoundedSource;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+
+public class CassandraReadIO {
+
+	static class Source extends BoundedSource{
+		CassandraReadConfiguration configuration;
+
+		Source(CassandraReadConfiguration config) throws IOException{
+			this.configuration = config;
+		}
+
+		@Override
+		public Coder<Entity> getDefaultOutputCoder() {
+			return (Coder<Entity>) SerializableCoder.of(configuration.get_entityName());
+		}
+
+		@Override
+		public void validate(){}
+		
+		private static class Reader<T> extends BoundedReader{
```

Comment:
Reading from Cassandra should be able to support dynamic work rebalancing, which will make this source much more efficient.  The javadoc on BoundedSource has good information on how to do that.  Absent that, you should at least provide a reasonable estimate of getEstimatedSizeBytes; otherwise, runners will have no way of choosing a reasonable number of initial bundles.